File: Java/src/main/java/domain/Reservation.java
package domain;

import java.util.List;

public class Reservation {
    public final TrainId trainId;
    public final String bookingId;
    public final List<Seat> seats;

    public Reservation(TrainId trainId, String bookingId, List<Seat> seats) {
        this.trainId = trainId;
        this.bookingId = bookingId;
        this.seats = seats;
    }
}

File: csharp/src/Domain/Reservation.cs
using System.Collections.Generic;

namespace TrainKata.Domain;

public struct Reservation
{
    public TrainId TrainId { get; }
    public BookingReference BookingReference { get; }
    public List<SeatId> Seats { get; }

    public Reservation(TrainId trainId, BookingReference bookingReference, List<SeatId> seats)
    {
        TrainId = trainId;
        BookingReference = bookingReference;
        Seats = seats;
    }
}